product-backend/src/test/java/org/kainos/ea/service/AddJobRoleServiceTest.java

```diff
@@ -0,0 +1,60 @@
+package org.kainos.ea.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.kainos.ea.api.JobRoleService;
+import org.kainos.ea.cli.AddJobRole;
+import org.kainos.ea.client.FailedToCreateJobRoleException;
+import org.kainos.ea.db.DatabaseConnector;
+import org.kainos.ea.db.JobRoleDao;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.sql.SQLException;
+import java.util.OptionalInt;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+
+@ExtendWith(MockitoExtension.class)
+public class AddJobRoleServiceTest {
+
+    JobRoleDao jobRoleDao = mock(JobRoleDao.class);
+    DatabaseConnector databaseConnector = mock(DatabaseConnector.class);
+    JobRoleService jobRoleService = new JobRoleService(jobRoleDao);
+    AddJobRole addJobRole = new AddJobRole(
+            "tomekk",
+            "test",
+            "Bloggs"
+    );
+
+    @Test
+    void createJobRole_shouldReturnId_whenDaoReturnsId() throws SQLException, FailedToCreateJobRoleException {
+        int expectedResult = 1;
+        Mockito.when(jobRoleDao.createRole(addJobRole)).thenReturn(OptionalInt.of(expectedResult));
+
+        int result = jobRoleService.createJobRole(addJobRole);
+
```

Comment:
You should verify mocks have been called with proper parameters and expected amount of calls.
```suggestion
        int result = jobRoleService.createJobRole(addJobRole);
        verify(jobRoleDao).createRole(addJobRole);
        verify(jobRoleDao).createSpec(addJobRole);

```